Fix unit tests

Potential steps/solutions:
- [ ] Implement virtual environment (with goatools, ete3, etc..)
- [x] Implement metaquantome DB into unit tests


```
> python -m unittest discover tests
/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/go-basic.obo: fmt(1.2) rel(2018-10-30) 47,358 GO Terms
/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/goslim_metagenomics.obo: fmt(1.2) rel(None) 185 GO Terms
/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/go-basic.obo: fmt(1.2) rel(2018-10-30) 47,358 GO Terms
/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/goslim_metagenomics.obo: fmt(1.2) rel(None) 185 GO Terms
ERROR:root:NCBI files not found in specified directory. Use metaquantome db to download files
ERROR:root:NCBI files not found in specified directory. Use metaquantome db to download files
/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/go-basic.obo: fmt(1.2) rel(2018-10-30) 47,358 GO Terms
/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/goslim_metagenomics.obo: fmt(1.2) rel(None) 185 GO Terms
/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/tmp_go_data_dwnld/go-basic.obo: fmt(1.2) rel(2021-01-01) 47,285 GO Terms
/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/tmp_go_data_dwnld/goslim_metagenomics.obo: fmt(1.2) rel(None) 201 GO Terms
.NCBI database format is outdated. Upgrading
Downloading taxdump.tar.gz from NCBI FTP site (via HTTP)...
Done. Parsing...
Loading node names...
2304656 names loaded.
237795 synonyms loaded.
Loading nodes...
2304656 nodes loaded.
Linking nodes...
Tree is loaded.
Updating database: /Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/tmp_test_tax_dwnld/taxa.sqlite ...
 2304000 generating entries... 
Uploading to /Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/tmp_test_tax_dwnld/taxa.sqlite

Inserting synonyms:      235000 
Inserting taxid merges:  60000 
Inserting taxids:       2300000 
.....300
./Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/go-basic.obo: fmt(1.2) rel(2018-10-30) 47,358 GO Terms
./Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/go-basic.obo: fmt(1.2) rel(2018-10-30) 47,358 GO Terms
./Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/go-basic.obo: fmt(1.2) rel(2018-10-30) 47,358 GO Terms
.ERROR:root:NCBI files not found in specified directory. Use metaquantome db to download files
EERROR:root:NCBI files not found in specified directory. Use metaquantome db to download files
EERROR:root:NCBI files not found in specified directory. Use metaquantome db to download files
E..E......../Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/go-basic.obo: fmt(1.2) rel(2018-10-30) 47,358 GO Terms
ERROR:root:NCBI files not found in specified directory. Use metaquantome db to download files
E../Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/go-basic.obo: fmt(1.2) rel(2018-10-30) 47,358 GO Terms
./Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/go-basic.obo: fmt(1.2) rel(2018-10-30) 47,358 GO Terms
/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/go-basic.obo: fmt(1.2) rel(2018-10-30) 47,358 GO Terms
../Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/go-basic.obo: fmt(1.2) rel(2018-10-30) 47,358 GO Terms
./Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/go-basic.obo: fmt(1.2) rel(2018-10-30) 47,358 GO Terms
/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/goslim_metagenomics.obo: fmt(1.2) rel(None) 185 GO Terms
.ERROR:root:NCBI files not found in specified directory. Use metaquantome db to download files
EERROR:root:NCBI files not found in specified directory. Use metaquantome db to download files
EERROR:root:NCBI files not found in specified directory. Use metaquantome db to download files
EERROR:root:NCBI files not found in specified directory. Use metaquantome db to download files
EERROR:root:NCBI files not found in specified directory. Use metaquantome db to download files
EERROR:root:NCBI files not found in specified directory. Use metaquantome db to download files
E...............EWarning message:
package ‘jsonlite’ was built under R version 3.6.2 
Warning message:
package ‘scico’ was built under R version 3.6.2 
Warning message:
package ‘jsonlite’ was built under R version 3.6.2 
Warning message:
package ‘scico’ was built under R version 3.6.2 
.Warning message:
package ‘jsonlite’ was built under R version 3.6.2 
Warning message:
package ‘scico’ was built under R version 3.6.2 
Warning message:
package ‘jsonlite’ was built under R version 3.6.2 
Warning message:
package ‘scico’ was built under R version 3.6.2 
Warning message:
package ‘jsonlite’ was built under R version 3.6.2 
Warning message:
package ‘scico’ was built under R version 3.6.2 
.Warning message:
package ‘jsonlite’ was built under R version 3.6.2 
Warning message:
package ‘scico’ was built under R version 3.6.2 
Warning message:
package ‘jsonlite’ was built under R version 3.6.2 
Warning message:
package ‘scico’ was built under R version 3.6.2 
Warning message:
package ‘jsonlite’ was built under R version 3.6.2 
Warning message:
package ‘scico’ was built under R version 3.6.2 
.Warning message:
package ‘jsonlite’ was built under R version 3.6.2 
Warning message:
package ‘scico’ was built under R version 3.6.2 
Error in `[.data.frame`(df, , fc_corr_p) : undefined columns selected
Calls: main -> volcano_cli -> mq_volcano -> [ -> [.data.frame
Execution halted
EWarning message:
package ‘jsonlite’ was built under R version 3.6.2 
Warning message:
package ‘scico’ was built under R version 3.6.2 
.Warning message:
package ‘jsonlite’ was built under R version 3.6.2 
Warning message:
package ‘scico’ was built under R version 3.6.2 
.Warning message:
package ‘jsonlite’ was built under R version 3.6.2 
Warning message:
package ‘scico’ was built under R version 3.6.2 
.Warning message:
package ‘jsonlite’ was built under R version 3.6.2 
Warning message:
package ‘scico’ was built under R version 3.6.2 
Error in `[.data.frame`(df, , fc_corr_p) : undefined columns selected
Calls: main -> volcano_cli -> mq_volcano -> [ -> [.data.frame
Execution halted
EE...E.....ERROR:root:NCBI files not found in specified directory. Use metaquantome db to download files
E..
======================================================================
ERROR: testAggregateNodes (travis.testAnnotationHierarchy.TestAnnotationHierarchyNcbi)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testAnnotationHierarchy.py", line 43, in testAggregateNodes
    ah, sample_set = self._create_sapiens_db()
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testAnnotationHierarchy.py", line 15, in _create_sapiens_db
    db = NCBITaxonomyDb(TEST_DIR)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/databases/NCBITaxonomyDb.py", line 51, in __init__
    self.ncbi = self._load_ncbi_db(data_dir)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/databases/NCBITaxonomyDb.py", line 100, in _load_ncbi_db
    return NCBITaxa(tax_path)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/venv/lib/python3.7/site-packages/ete3/ncbi_taxonomy/ncbiquery.py", line 115, in __init__
    raise ValueError("Cannot open taxonomy database: %s" % self.dbfile)
ValueError: Cannot open taxonomy database: /Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/taxa.sqlite

======================================================================
ERROR: testInit (travis.testAnnotationHierarchy.TestAnnotationHierarchyNcbi)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testAnnotationHierarchy.py", line 21, in testInit
    ah, sample_set = self._create_sapiens_db()
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testAnnotationHierarchy.py", line 15, in _create_sapiens_db
    db = NCBITaxonomyDb(TEST_DIR)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/databases/NCBITaxonomyDb.py", line 51, in __init__
    self.ncbi = self._load_ncbi_db(data_dir)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/databases/NCBITaxonomyDb.py", line 100, in _load_ncbi_db
    return NCBITaxa(tax_path)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/venv/lib/python3.7/site-packages/ete3/ncbi_taxonomy/ncbiquery.py", line 115, in __init__
    raise ValueError("Cannot open taxonomy database: %s" % self.dbfile)
ValueError: Cannot open taxonomy database: /Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/taxa.sqlite

======================================================================
ERROR: testUpdateNode (travis.testAnnotationHierarchy.TestAnnotationHierarchyNcbi)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testAnnotationHierarchy.py", line 27, in testUpdateNode
    ah, sample_set = self._create_sapiens_db()
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testAnnotationHierarchy.py", line 15, in _create_sapiens_db
    db = NCBITaxonomyDb(TEST_DIR)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/databases/NCBITaxonomyDb.py", line 51, in __init__
    self.ncbi = self._load_ncbi_db(data_dir)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/databases/NCBITaxonomyDb.py", line 100, in _load_ncbi_db
    return NCBITaxa(tax_path)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/venv/lib/python3.7/site-packages/ete3/ncbi_taxonomy/ncbiquery.py", line 115, in __init__
    raise ValueError("Cannot open taxonomy database: %s" % self.dbfile)
ValueError: Cannot open taxonomy database: /Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/taxa.sqlite

======================================================================
ERROR: travis.testCLI (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: travis.testCLI
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testCLI.py", line 104
    '--fc_corr_p', ,
                   ^
SyntaxError: invalid syntax


======================================================================
ERROR: testDifferentNames (travis.testExpand.TestFunctionTaxonomyAnalysis)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testExpand.py", line 190, in testDifferentNames
    tax_colname='lca', func_file=func, func_colname="go")
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/modules/expand.py", line 59, in expand
    samp_grps=samp_grps, ft_tar_rank=ft_tar_rank, data_dir=data_dir)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/modules/function_taxonomy_interaction.py", line 56, in function_taxonomy_analysis
    ncbi = NCBITaxonomyDb(data_dir)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/databases/NCBITaxonomyDb.py", line 51, in __init__
    self.ncbi = self._load_ncbi_db(data_dir)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/databases/NCBITaxonomyDb.py", line 100, in _load_ncbi_db
    return NCBITaxa(tax_path)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/venv/lib/python3.7/site-packages/ete3/ncbi_taxonomy/ncbiquery.py", line 115, in __init__
    raise ValueError("Cannot open taxonomy database: %s" % self.dbfile)
ValueError: Cannot open taxonomy database: /Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/taxa.sqlite

======================================================================
ERROR: testMultCols (travis.testExpand.TestTaxonomyAnalysisExpand)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testExpand.py", line 153, in testMultCols
    tax_colname='lca')
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/modules/expand.py", line 55, in expand
    results = taxonomy_analysis(df=df, samp_grps=samp_grps, data_dir=data_dir, tax_colname=tax_colname)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/modules/taxonomy_analysis.py", line 20, in taxonomy_analysis
    ncbi = NCBITaxonomyDb(data_dir)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/databases/NCBITaxonomyDb.py", line 51, in __init__
    self.ncbi = self._load_ncbi_db(data_dir)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/databases/NCBITaxonomyDb.py", line 100, in _load_ncbi_db
    return NCBITaxa(tax_path)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/venv/lib/python3.7/site-packages/ete3/ncbi_taxonomy/ncbiquery.py", line 115, in __init__
    raise ValueError("Cannot open taxonomy database: %s" % self.dbfile)
ValueError: Cannot open taxonomy database: /Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/taxa.sqlite

======================================================================
ERROR: testNopep (travis.testExpand.TestTaxonomyAnalysisExpand)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testExpand.py", line 160, in testNopep
    nopep=True, nopep_file=nopep)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/modules/expand.py", line 55, in expand
    results = taxonomy_analysis(df=df, samp_grps=samp_grps, data_dir=data_dir, tax_colname=tax_colname)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/modules/taxonomy_analysis.py", line 20, in taxonomy_analysis
    ncbi = NCBITaxonomyDb(data_dir)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/databases/NCBITaxonomyDb.py", line 51, in __init__
    self.ncbi = self._load_ncbi_db(data_dir)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/databases/NCBITaxonomyDb.py", line 100, in _load_ncbi_db
    return NCBITaxa(tax_path)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/venv/lib/python3.7/site-packages/ete3/ncbi_taxonomy/ncbiquery.py", line 115, in __init__
    raise ValueError("Cannot open taxonomy database: %s" % self.dbfile)
ValueError: Cannot open taxonomy database: /Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/taxa.sqlite

======================================================================
ERROR: testParentIntensityHigher (travis.testExpand.TestTaxonomyAnalysisExpand)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testExpand.py", line 172, in testParentIntensityHigher
    tax_colname='taxon_id')
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/modules/expand.py", line 55, in expand
    results = taxonomy_analysis(df=df, samp_grps=samp_grps, data_dir=data_dir, tax_colname=tax_colname)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/modules/taxonomy_analysis.py", line 20, in taxonomy_analysis
    ncbi = NCBITaxonomyDb(data_dir)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/databases/NCBITaxonomyDb.py", line 51, in __init__
    self.ncbi = self._load_ncbi_db(data_dir)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/databases/NCBITaxonomyDb.py", line 100, in _load_ncbi_db
    return NCBITaxa(tax_path)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/venv/lib/python3.7/site-packages/ete3/ncbi_taxonomy/ncbiquery.py", line 115, in __init__
    raise ValueError("Cannot open taxonomy database: %s" % self.dbfile)
ValueError: Cannot open taxonomy database: /Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/taxa.sqlite

======================================================================
ERROR: testSingleBasic (travis.testExpand.TestTaxonomyAnalysisExpand)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testExpand.py", line 135, in testSingleBasic
    tax_colname='lca')
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/modules/expand.py", line 55, in expand
    results = taxonomy_analysis(df=df, samp_grps=samp_grps, data_dir=data_dir, tax_colname=tax_colname)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/modules/taxonomy_analysis.py", line 20, in taxonomy_analysis
    ncbi = NCBITaxonomyDb(data_dir)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/databases/NCBITaxonomyDb.py", line 51, in __init__
    self.ncbi = self._load_ncbi_db(data_dir)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/databases/NCBITaxonomyDb.py", line 100, in _load_ncbi_db
    return NCBITaxa(tax_path)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/venv/lib/python3.7/site-packages/ete3/ncbi_taxonomy/ncbiquery.py", line 115, in __init__
    raise ValueError("Cannot open taxonomy database: %s" % self.dbfile)
ValueError: Cannot open taxonomy database: /Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/taxa.sqlite

======================================================================
ERROR: testWrite (travis.testExpand.TestTaxonomyAnalysisExpand)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testExpand.py", line 144, in testWrite
    tax_colname='lca')
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/modules/expand.py", line 55, in expand
    results = taxonomy_analysis(df=df, samp_grps=samp_grps, data_dir=data_dir, tax_colname=tax_colname)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/modules/taxonomy_analysis.py", line 20, in taxonomy_analysis
    ncbi = NCBITaxonomyDb(data_dir)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/databases/NCBITaxonomyDb.py", line 51, in __init__
    self.ncbi = self._load_ncbi_db(data_dir)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/databases/NCBITaxonomyDb.py", line 100, in _load_ncbi_db
    return NCBITaxa(tax_path)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/venv/lib/python3.7/site-packages/ete3/ncbi_taxonomy/ncbiquery.py", line 115, in __init__
    raise ValueError("Cannot open taxonomy database: %s" % self.dbfile)
ValueError: Cannot open taxonomy database: /Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/taxa.sqlite

======================================================================
ERROR: testFilter (travis.testFilter.TestFilter)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testFilter.py", line 25, in testFilter
    tax_colname='lca')
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/modules/expand.py", line 55, in expand
    results = taxonomy_analysis(df=df, samp_grps=samp_grps, data_dir=data_dir, tax_colname=tax_colname)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/modules/taxonomy_analysis.py", line 20, in taxonomy_analysis
    ncbi = NCBITaxonomyDb(data_dir)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/databases/NCBITaxonomyDb.py", line 51, in __init__
    self.ncbi = self._load_ncbi_db(data_dir)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/databases/NCBITaxonomyDb.py", line 100, in _load_ncbi_db
    return NCBITaxa(tax_path)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/venv/lib/python3.7/site-packages/ete3/ncbi_taxonomy/ncbiquery.py", line 115, in __init__
    raise ValueError("Cannot open taxonomy database: %s" % self.dbfile)
ValueError: Cannot open taxonomy database: /Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/taxa.sqlite

======================================================================
ERROR: travis.testNCBITaxonomyDb (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: travis.testNCBITaxonomyDb
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testNCBITaxonomyDb.py", line 11, in <module>
    class TestTaxonomyDatabase(unittest.TestCase):
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testNCBITaxonomyDb.py", line 12, in TestTaxonomyDatabase
    ncbi = td.NCBITaxonomyDb(TEST_DIR)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/databases/NCBITaxonomyDb.py", line 51, in __init__
    self.ncbi = self._load_ncbi_db(data_dir)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/databases/NCBITaxonomyDb.py", line 100, in _load_ncbi_db
    return NCBITaxa(tax_path)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/venv/lib/python3.7/site-packages/ete3/ncbi_taxonomy/ncbiquery.py", line 115, in __init__
    raise ValueError("Cannot open taxonomy database: %s" % self.dbfile)
ValueError: Cannot open taxonomy database: /Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/taxa.sqlite


======================================================================
ERROR: testGOVolcano (travis.testRunViz.TestRunViz)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testRunViz.py", line 72, in testGOVolcano
    gosplit=True)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/modules/run_viz.py", line 53, in run_viz
    subprocess.run(cmd_string, stdout=fnull, check=True)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/subprocess.py", line 487, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['Rscript', '--vanilla', '/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/modules/viz.R', 'volcano', '/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/test.png', '/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/go_tested.tab', 'id', 'log2fc_s1_over_s2', 'None', 'False', 'True', '5', '5', 'None']' returned non-zero exit status 1.

======================================================================
ERROR: testVolcano (travis.testRunViz.TestRunViz)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testRunViz.py", line 56, in testVolcano
    fc_name="log2fc_s1_over_s2")
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/modules/run_viz.py", line 53, in run_viz
    subprocess.run(cmd_string, stdout=fnull, check=True)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/subprocess.py", line 487, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['Rscript', '--vanilla', '/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/modules/viz.R', 'volcano', '/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/test.png', '/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/cli_mult_test_out.tab', 'id', 'log2fc_s1_over_s2', 'None', 'False', 'False', '5', '5', 'None']' returned non-zero exit status 1.

======================================================================
ERROR: travis.testSampleAnnotations (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: travis.testSampleAnnotations
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testSampleAnnotations.py", line 10, in <module>
    class TestSampleAnnotations(unittest.TestCase):
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testSampleAnnotations.py", line 11, in TestSampleAnnotations
    ncbi = NCBITaxonomyDb(TEST_DIR)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/databases/NCBITaxonomyDb.py", line 51, in __init__
    self.ncbi = self._load_ncbi_db(data_dir)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/databases/NCBITaxonomyDb.py", line 100, in _load_ncbi_db
    return NCBITaxa(tax_path)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/venv/lib/python3.7/site-packages/ete3/ncbi_taxonomy/ncbiquery.py", line 115, in __init__
    raise ValueError("Cannot open taxonomy database: %s" % self.dbfile)
ValueError: Cannot open taxonomy database: /Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/taxa.sqlite


======================================================================
ERROR: travis.testStat (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: travis.testStat
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testStat.py", line 107
    outfile=None)
          ^
SyntaxError: invalid syntax


======================================================================
ERROR: testFilter (travis.testUtils.TestUtils)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testUtils.py", line 27, in testFilter
    ncbi_db = ncbi.NCBITaxonomyDb(TEST_DIR)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/databases/NCBITaxonomyDb.py", line 51, in __init__
    self.ncbi = self._load_ncbi_db(data_dir)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/databases/NCBITaxonomyDb.py", line 100, in _load_ncbi_db
    return NCBITaxa(tax_path)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/venv/lib/python3.7/site-packages/ete3/ncbi_taxonomy/ncbiquery.py", line 115, in __init__
    raise ValueError("Cannot open taxonomy database: %s" % self.dbfile)
ValueError: Cannot open taxonomy database: /Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/taxa.sqlite

----------------------------------------------------------------------
Ran 75 tests in 210.393s

FAILED (errors=17)
```

Downloaded the NCBI databases prior to testing as follows:
```
metaquantome db ncbi --dir ./metaquantome/data/test/
```

Errors are now reduced
```
/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/go-basic.obo: fmt(1.2) rel(2018-10-30) 47,358 GO Terms
/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/goslim_metagenomics.obo: fmt(1.2) rel(None) 185 GO Terms
/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/go-basic.obo: fmt(1.2) rel(2018-10-30) 47,358 GO Terms
/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/goslim_metagenomics.obo: fmt(1.2) rel(None) 185 GO Terms
/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/go-basic.obo: fmt(1.2) rel(2018-10-30) 47,358 GO Terms
/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/goslim_metagenomics.obo: fmt(1.2) rel(None) 185 GO Terms
/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/tmp_go_data_dwnld/go-basic.obo: fmt(1.2) rel(2021-01-01) 47,285 GO Terms
/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/tmp_go_data_dwnld/goslim_metagenomics.obo: fmt(1.2) rel(None) 201 GO Terms
.NCBI database format is outdated. Upgrading
Downloading taxdump.tar.gz from NCBI FTP site (via HTTP)...
Done. Parsing...
Loading node names...
2306240 names loaded.
240376 synonyms loaded.
Loading nodes...
2306240 nodes loaded.
Linking nodes...
Tree is loaded.
Updating database: /Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/tmp_test_tax_dwnld/taxa.sqlite ...
 2306000 generating entries... 
Uploading to /Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/tmp_test_tax_dwnld/taxa.sqlite

Inserting synonyms:      240000 
Inserting taxid merges:  60000 
Inserting taxids:       2305000 
.....300
./Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/go-basic.obo: fmt(1.2) rel(2018-10-30) 47,358 GO Terms
./Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/go-basic.obo: fmt(1.2) rel(2018-10-30) 47,358 GO Terms
./Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/go-basic.obo: fmt(1.2) rel(2018-10-30) 47,358 GO Terms
......E......../Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/go-basic.obo: fmt(1.2) rel(2018-10-30) 47,358 GO Terms
E../Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/go-basic.obo: fmt(1.2) rel(2018-10-30) 47,358 GO Terms
./Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/go-basic.obo: fmt(1.2) rel(2018-10-30) 47,358 GO Terms
/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/go-basic.obo: fmt(1.2) rel(2018-10-30) 47,358 GO Terms
../Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/go-basic.obo: fmt(1.2) rel(2018-10-30) 47,358 GO Terms
./Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/go-basic.obo: fmt(1.2) rel(2018-10-30) 47,358 GO Terms
/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/goslim_metagenomics.obo: fmt(1.2) rel(None) 185 GO Terms
..............................F....Warning message:
package ‘jsonlite’ was built under R version 3.6.2 
Warning message:
package ‘scico’ was built under R version 3.6.2 
Warning message:
package ‘jsonlite’ was built under R version 3.6.2 
Warning message:
package ‘scico’ was built under R version 3.6.2 
.Warning message:
package ‘jsonlite’ was built under R version 3.6.2 
Warning message:
package ‘scico’ was built under R version 3.6.2 
Warning message:
package ‘jsonlite’ was built under R version 3.6.2 
Warning message:
package ‘scico’ was built under R version 3.6.2 
Warning message:
package ‘jsonlite’ was built under R version 3.6.2 
Warning message:
package ‘scico’ was built under R version 3.6.2 
.Warning message:
package ‘jsonlite’ was built under R version 3.6.2 
Warning message:
package ‘scico’ was built under R version 3.6.2 
Warning message:
package ‘jsonlite’ was built under R version 3.6.2 
Warning message:
package ‘scico’ was built under R version 3.6.2 
Warning message:
package ‘jsonlite’ was built under R version 3.6.2 
Warning message:
package ‘scico’ was built under R version 3.6.2 
.Warning message:
package ‘jsonlite’ was built under R version 3.6.2 
Warning message:
package ‘scico’ was built under R version 3.6.2 
Error in `[.data.frame`(df, , fc_corr_p) : undefined columns selected
Calls: main -> volcano_cli -> mq_volcano -> [ -> [.data.frame
Execution halted
EWarning message:
package ‘jsonlite’ was built under R version 3.6.2 
Warning message:
package ‘scico’ was built under R version 3.6.2 
.Warning message:
package ‘jsonlite’ was built under R version 3.6.2 
Warning message:
package ‘scico’ was built under R version 3.6.2 
.Warning message:
package ‘jsonlite’ was built under R version 3.6.2 
Warning message:
package ‘scico’ was built under R version 3.6.2 
.Warning message:
package ‘jsonlite’ was built under R version 3.6.2 
Warning message:
package ‘scico’ was built under R version 3.6.2 
Error in `[.data.frame`(df, , fc_corr_p) : undefined columns selected
Calls: main -> volcano_cli -> mq_volcano -> [ -> [.data.frame
Execution halted
E.......E........
======================================================================
ERROR: travis.testCLI (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: travis.testCLI
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testCLI.py", line 104
    '--fc_corr_p', ,
                   ^
SyntaxError: invalid syntax


======================================================================
ERROR: testDifferentNames (travis.testExpand.TestFunctionTaxonomyAnalysis)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testExpand.py", line 190, in testDifferentNames
    tax_colname='lca', func_file=func, func_colname="go")
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/modules/expand.py", line 59, in expand
    samp_grps=samp_grps, ft_tar_rank=ft_tar_rank, data_dir=data_dir)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/modules/function_taxonomy_interaction.py", line 78, in function_taxonomy_analysis
    grouped = df_int.groupby(by=[func_colname, 'des_rank']).sum(axis=1)
TypeError: sum() got an unexpected keyword argument 'axis'

======================================================================
ERROR: testGOVolcano (travis.testRunViz.TestRunViz)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testRunViz.py", line 72, in testGOVolcano
    gosplit=True)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/modules/run_viz.py", line 53, in run_viz
    subprocess.run(cmd_string, stdout=fnull, check=True)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/subprocess.py", line 487, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['Rscript', '--vanilla', '/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/modules/viz.R', 'volcano', '/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/test.png', '/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/go_tested.tab', 'id', 'log2fc_s1_over_s2', 'None', 'False', 'True', '5', '5', 'None']' returned non-zero exit status 1.

======================================================================
ERROR: testVolcano (travis.testRunViz.TestRunViz)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testRunViz.py", line 56, in testVolcano
    fc_name="log2fc_s1_over_s2")
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/modules/run_viz.py", line 53, in run_viz
    subprocess.run(cmd_string, stdout=fnull, check=True)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/subprocess.py", line 487, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['Rscript', '--vanilla', '/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/modules/viz.R', 'volcano', '/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/test.png', '/Users/rsajulga/Projects/umn/galaxyp/metaquantome/metaquantome/data/test/cli_mult_test_out.tab', 'id', 'log2fc_s1_over_s2', 'None', 'False', 'False', '5', '5', 'None']' returned non-zero exit status 1.

======================================================================
ERROR: travis.testStat (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: travis.testStat
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testStat.py", line 107
    outfile=None)
          ^
SyntaxError: invalid syntax


======================================================================
FAIL: testGetDescendants (travis.testNCBITaxonomyDb.TestTaxonomyDatabase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/rsajulga/Projects/umn/galaxyp/metaquantome/tests/travis/testNCBITaxonomyDb.py", line 68, in testGetDescendants
    self.assertSetEqual(self.ncbi.get_descendants(id), descendants_exp)
AssertionError: Items in the first set but not the second:
2665953

----------------------------------------------------------------------
Ran 90 tests in 275.652s

FAILED (failures=1, errors=5)
```

--- a/README.md
+++ b/README.md
@@ -69,13 +69,15 @@ In-depth tutorials can be found [here](https://galaxyproteomics.github.io/metaqu
 2. [metaQuantome Galaxy tool](https://galaxyproteomics.github.io/metaquantome_mcp_analysis/galaxy_tutorial/galaxy_tutorial.html)
 
 ## Tests [⤴](#table-of-contents)
+
+
+For testing, you will also need to set up a BioConda environment as specified [above](#bioconda-).
+
 Before running unittests for the project, you will need to install some databases as follows:
 ```
 metaquantome db ncbi --dir ./metaquantome/data/test/
 ```
 *Note: This step will take a few minutes to download*
-
-You will also need to set up a BioConda environment as specified [above](#bioconda-).
 
 Once set up, run the following from the root directory:
 

--- a/README.md
+++ b/README.md
@@ -5,28 +5,81 @@
 # metaQuantome
 
 
-Quantitative analysis of the function and taxonomy of microbiomes and their interaction.
+Quantitative analysis on the function and taxonomy of microbiomes and their interaction.
 
+## Table of Contents
+- [Setup](#setup-)
+    - [BioConda](#bioconda-)
+    - [Local Development](#local-development-)
+- [Usage](#usage-)
+- [Tests](#tests-)
 
-# Developers
+## Setup [⤴](#table-of-contents)
 
-## Setup
+### BioConda
 
-The newest version of metaquantome should be downloaded from this site.
-The dependencies are most easily satisfied with conda, and the environment can
-be created and activated as follows:
+The easiest way to install metaQuantome with all the dependencies is by using Bioconda (provided you are on Mac or Linux, which are the only systems supported by Bioconda).
 
-```sh
-conda env update --file dev_environment.yml
-source activate metaquantome
+First, install the conda package manager, by downloading either Anaconda or Miniconda (see https://docs.anaconda.com/anaconda/install/). Then, the following commands will set up the necessary channels for Bioconda and metaQuantome. If needed, additional information on Bioconda is available at https://bioconda.github.io/.
+
+```
+conda config --add channels defaults
+conda config --add channels bioconda
+conda config --add channels conda-forge
 ```
 
-Note that the bioconda and conda forge channels must be enabled,
-as described on [the bioconda website](https://bioconda.github.io/#set-up-channels).
+Then, run the following command to set up an environment named mqome, which will have metaQuantome (version 2.0.0) and all dependencies in it:
 
-## Tests
-To run unittests for the project, run the following from the root directory:
+```
+conda create -n mqome metaquantome=2.0.0
+```
+
+If the following prompt is seen at the command line, type `y`:
+```
+Proceeed ([y]/n)?
+```
+
+The create command only needs to be run once.
+
+Finally, we can activate the environment using the following command, which will make the `metaquantome` command available.
+```
+conda activate mqome
+```
+
+### Local Development
+
+To begin, ensure that you have Python3 installed. To check, issue this command to verify your python version:
+```
+python --version
+```
+
+If Python3 is not installed, please download it from [here](https://www.python.org/downloads/).
+
+Pip is the package installer for Python. It comes pre-packaged with Python. 
+
+To install your local version of metaQuantome, run the following in the root directory:
+```
+pip install .
+```
+
+## Usage [⤴](#table-of-contents)
+
+In-depth tutorials can be found [here](https://galaxyproteomics.github.io/metaquantome_mcp_analysis/) for the following:
+1. [metaQuantome command-line interface](https://galaxyproteomics.github.io/metaquantome_mcp_analysis/cli_tutorial/cli_tutorial.html)
+2. [metaQuantome Galaxy tool](https://galaxyproteomics.github.io/metaquantome_mcp_analysis/galaxy_tutorial/galaxy_tutorial.html)
+
+## Tests [⤴](#table-of-contents)
+Before running unittests for the project, you will need to install some databases as follows:
+```
+metaquantome db ncbi --dir ./metaquantome/data/test/
+```
+*Note: This step will take a few minutes to download*
+
+You will also need to set up a BioConda environment as specified [above](#bioconda-).
+
+Once set up, run the following from the root directory:
 
 ```sh
 python -m unittest discover tests
 ```
+*Note: This step will take a few minutes to run*

--- a/metaquantome/cli.py
+++ b/metaquantome/cli.py
@@ -2,6 +2,7 @@ import sys
 import argparse
 import logging
 import os
+import pkg_resources
 
 # add metaquantome parent directory to path
 sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
@@ -88,6 +89,9 @@ def parse_args_cli():
     :return: parsed arguments
     """
     parser = argparse.ArgumentParser()
+    parser.add_argument('-v', '--version', action='version',
+                         version=pkg_resources.require("metaquantome")[0].version)
+
 
     # split this into three submodules
     subparsers = parser.add_subparsers(title="commands", dest="command")
@@ -290,6 +294,9 @@ def parse_args_cli():
     pca.add_argument("--calculate_sep", action="store_true",
                      help="Flag. Calculate separation between groups and include in title?")
 
+    if len(sys.argv)==1:
+        parser.print_help(sys.stderr)
+        sys.exit(1)
     args = parser.parse_args()
     return args
 

--- a/metaquantome/cli.py
+++ b/metaquantome/cli.py
@@ -88,18 +88,70 @@ def parse_args_cli():
     parse the command line arguments
     :return: parsed arguments
     """
-    parser = argparse.ArgumentParser()
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.RawTextHelpFormatter,
+        description="metaQuantome is a tool that performs quantitative analysis"
+            " on the function and taxonomy of microbomes and their interactions. "
+            "For more background information, please read the associated manuscript: https://doi.org/10.1074/mcp.ra118.001240. "
+            "For a more hands-on tutorial, please visit the following page: https://galaxyproteomics.github.io/metaquantome_mcp_analysis/cli_tutorial/cli_tutorial.html.\n\n"
+            "The metaQuantome workflow is as follows: db → expand → filter → stat → viz.\n\n" 
+            "Run `metaquantome {db,expand,filter,stat,viz} -h` for more information on the individual modules. "
+            "Any issues can be brought to attention here: https://github.com/galaxyproteomics/metaquantome/issues."
+    )
     parser.add_argument('-v', '--version', action='version',
                          version=pkg_resources.require("metaquantome")[0].version)
 
-
     # split this into three submodules
     subparsers = parser.add_subparsers(title="commands", dest="command")
-    parser_db = subparsers.add_parser('db')
-    parser_expand = subparsers.add_parser('expand')
-    parser_filter = subparsers.add_parser('filter')
-    parser_stat = subparsers.add_parser('stat')
-    parser_viz = subparsers.add_parser('viz')
+    parser_db = subparsers.add_parser('db',
+        formatter_class=argparse.RawTextHelpFormatter,
+        description="metaQuantome uses freely available bioinformatic databases to expand your set of direct annotations. For most cases, all 3 databases can be downloaded (the default).\n"
+                    "The databases are:\n"
+                    "1. NCBI taxonomy database. This contains a list of all currently identified taxa and the relationships between them.\n"
+                    "2. Gene Ontology (GO) term database. metaQuantome uses the OBO format of the database. Specifically, two files are used: the go-basic.obo file, which is a simplified version of the GO database that is guaranteed to be acyclic, and the metagenomics slim GO, which is a subset of the full GO that is useful for microbiome research. More details are available at http://geneontology.org/docs/download-ontology/\n"
+                    "3. ENZYME database with Enzyme Classification (EC) numbers. This database classifies enzymes and organizes the relationships between them.\n"
+                    "This module downloads the most recent releases of the specified databases and stores them in a single file, which can then be accessed by the rest of the metaQuantome modules. For reference, the taxonomy database is the largest (~500 Mb), while the GO and EC databases are smaller: ~34 Mb and ~10Mb, respectively."
+                    )
+    parser_expand = subparsers.add_parser('expand', 
+        formatter_class=argparse.RawTextHelpFormatter,
+        description='The expand module is the first analysis step in the metaQuantome analysis workflow, and can be run to analyze function, taxonomy, or function and taxonomy together.\n'
+                    'To prepare to run this module, you must create your samples file (tabular file with `group` and `colnames` columns with each row as a sample group) and download the necessary databases with `metaquantome db`.\n'
+                    'Some example analysis workflows are:\n'
+                    '1. Get the functional, taxonomic, or functional-taxonomic distribution: run expand, filter, and viz.\n'
+                    '2. Cluster analysis: run expand, filter, and viz. The viz module has heatmaps and PCA plots for cluster analysis.\n'
+                    '3. Differential expression: run expand, filter, stat, and viz.\n\n'
+                    'The following information is required for all 3 analysis modes (function, taxonomy, and function-taxonomy).\n'
+                    '- experimental design information.\n'
+                    '- a tab-separated peptide intensity file.\n'
+                    '- the name of the peptide column in the intensity file.\n\n'
+                    'Function mode\n'
+                    'In function mode, the following information is required:\n'
+                    '- the ontology being used: Gene Ontology (GO), Clusters of Orthologous Groups (COG), or Enzyme Commission (EC) numbers.\n'
+                    '- a tab-separated functional annotation file, with a peptide column and a functional annotation column. An entry in the functional annotation column may contain multiple functional annotations separated by commas.\n'
+                    '- the name of the peptide column in the functional annotation file.\n'
+                    '- the name of the functional annotation column in the functional annotation file.\n\n'
+                    'Taxonomy mode\n'
+                    'In taxonomy mode, the following information is required:\n'
+                    '- a tab-separated taxonomy annotation file, with a peptide column and a taxonomy annotation column. The taxonomic annotations should be the lowest common ancestor (LCA) for each peptide, preferably given as NCBI taxonomy IDs.\n'
+                    '- the name of the peptide column in the taxonomic annotation file.\n'
+                    '- the name of the taxonomy annotation column in the taxonomy annotation file.\n\n'
+                    'Function-Taxonomy mode\n'
+                    'In the combined mode, all of the above must be provided. In addition, the "target rank" must be provided, which is the desired taxonomic rank at which to summarize the function/taxonomy results.')
+    parser_filter = subparsers.add_parser('filter',
+        formatter_class=argparse.RawTextHelpFormatter, 
+        description="The filter module is the second step in the metaQuantome analysis workflow. The purpose of the filter module is to filter expanded terms to those that are representative and well-supported by the data. "
+                "Please see the manuscript (https://doi.org/10.1074/mcp.ra118.001240) for further details about filtering.")
+    parser_stat = subparsers.add_parser('stat', 
+        formatter_class=argparse.RawTextHelpFormatter, 
+        description="The stat module is the third step in the metaQuantome analysis workflow."
+                    " The purpose of the stat module is to perform differential expression analysis between 2 experimental conditions."
+                    " metaQuantome offers paired and unpaired tests, as well as parametric and non-parametric options.")
+    parser_viz = subparsers.add_parser('viz', 
+        formatter_class=argparse.RawTextHelpFormatter, 
+        description="The viz module is the final step in the metaQuantome analysis workflow."
+            " The available visualizations are:\n"
+                "-bar plot\n-volcano plot\n-heatmap\n-PCA plot\n"
+                "Please consult the manuscript (https://doi.org/10.1074/mcp.ra118.001240.) for full details on each of these plots.")
 
     # db download
     parser_db.add_argument('dbs', type=str, nargs='+', choices=['ncbi', 'go', 'ec'],

--- a/metaquantome/modules/function_taxonomy_interaction.py
+++ b/metaquantome/modules/function_taxonomy_interaction.py
@@ -75,10 +75,10 @@ def function_taxonomy_analysis(df, func_colname, pep_colname, ontology, slim_dow
     df_counts = df_int.copy()
     df_counts.loc[:, samp_grps.all_intcols] = df_counts.loc[:, samp_grps.all_intcols] > 0
     # group by both cog and lca and add
-    grouped = df_int.groupby(by=[func_colname, 'des_rank']).sum(axis=1)
+    grouped = df_int.groupby(by=[func_colname, 'des_rank']).sum()
     # get groupwise counts (i.e., unique peptides)
     # multiply by 1 to convert any single booleans (True) to 1
-    counts = df_counts.groupby(by=[func_colname, 'des_rank']).sum(axis=1) * 1
+    counts = df_counts.groupby(by=[func_colname, 'des_rank']).sum() * 1
     ints_and_counts = grouped.join(counts, rsuffix='_n_peptide')
 
     # ---- output prep ---- #

--- a/metaquantome/modules/viz.R
+++ b/metaquantome/modules/viz.R
@@ -452,13 +452,17 @@ prcomp_cli <- function(args){
 ####### ==================== #######
 #              VOLCANO             #
 ####### ==================== #######
-mq_volcano <- function(df, img, fc_name, fc_corr_p, flip_fc, width, height, textannot, gosplit, tabfile){
+mq_volcano <- function(df, img, fc_name, fc_corr_p, flip_fc, gosplit, width, height, textannot, tabfile){
     # df is the dataframe after stat
     # fc_name is the name of the column with the fold change data
     # textcol is the name of the column with the text describing the term
     df$fc <- df[, fc_name]
     if (flip_fc){
         df$fc <- (-1)*df$fc
+    }
+    if (fc_corr_p == "None"){
+        stop("corrected p-value column not defined. did you run metaquantome stat?",
+                call. = FALSE)
     }
     df$corrected_p <- df[, fc_corr_p]
     #df$neglog10p <- -log10(df[, "corrected_p"])

--- a/tests/travis/testCLI.py
+++ b/tests/travis/testCLI.py
@@ -100,13 +100,13 @@ class TestCLI(unittest.TestCase):
             '--infile', infile,
             '--img', imgfile,
             "--samps '", TTEST_SINFO, "'",
-            '--filter_to_sig', 'corrected_p_s1_over_s2',
-            '--fc_corr_p', ,
+            '--filter_to_sig', 
+            '--fc_corr_p', 'corrected_p_s1_over_s2',
             '--alpha 0.5'
         ])
         test_status = subprocess.call(cmd, shell=True)
         self.assertEqual(test_status, 0)
-        cmd2 = ' '.join([
+        cmd2 = ' '.join([ 
             'python3 metaquantome/cli.py viz -m f --ontology ec --plottype heatmap',
             '--infile', infile,
             '--img', imgfile,

--- a/tests/travis/testNCBITaxonomyDb.py
+++ b/tests/travis/testNCBITaxonomyDb.py
@@ -59,12 +59,12 @@ class TestTaxonomyDatabase(unittest.TestCase):
         # all descendants include 4 genuses/geni (9592, 9605, 9599, 9596)
         # and species:
         # gorilla species (499232, 9593)
-        # homo species (1425170, 9606)
+        # homo species (1425170, 9606, 2665953)
         # pan species (9597, 9598)
         # and pongo species (9601, 502961, 9600, 2051901, 9603)
         descendants_exp = {9592, 9605, 9599, 9596, 499232, 9593,
                            1425170, 9606, 9597, 9598,
-                           9601, 502961, 9600, 2051901, 9603}
+                           9601, 502961, 9600, 2051901, 9603, 2665953}
         self.assertSetEqual(self.ncbi.get_descendants(id), descendants_exp)
 
     def testGetParents(self):

--- a/tests/travis/testRunViz.py
+++ b/tests/travis/testRunViz.py
@@ -53,6 +53,7 @@ class TestRunViz(unittest.TestCase):
         infile = testfile('cli_mult_test_out.tab')
         run_viz('volcano', self.img, infile,
                 textannot="id",
+                fc_corr_p="corrected_p_s1_over_s2",
                 fc_name="log2fc_s1_over_s2")
 
         # test tabfile
@@ -60,6 +61,7 @@ class TestRunViz(unittest.TestCase):
         run_viz('volcano', self.img, infile,
                 textannot="id",
                 fc_name="log2fc_s1_over_s2",
+                fc_corr_p="corrected_p_s1_over_s2",
                 tabfile=tabfile)
         self.assertTrue(os.path.exists(tabfile))
         os.remove(tabfile)
@@ -69,6 +71,7 @@ class TestRunViz(unittest.TestCase):
         run_viz('volcano', self.img, infile,
                 textannot="id",
                 fc_name="log2fc_s1_over_s2",
+                fc_corr_p="corrected_p_s1_over_s2",
                 gosplit=True)
 
     def testHeatmap(self):

--- a/tests/travis/testStat.py
+++ b/tests/travis/testStat.py
@@ -65,6 +65,7 @@ class TestFunctionalAnalysisTest(unittest.TestCase):
         df_tst = stat.stat(expanded, sinfo=TTEST_SINFO, paired=False, parametric=True, ontology='go', mode='f', control_group='s2',
                            outfile=test_write)
         # make sure false is > 0.05 and trues are less than 0.05
+        df_tst.set_index('id', inplace=True)
         self.assertTrue(df_tst['p_s1_over_s2']['GO:0008152'] > 0.05)
         self.assertTrue(df_tst['p_s1_over_s2'][['GO:0022610','GO:0000003','GO:0032505']].le(0.05).all())
 
@@ -78,6 +79,7 @@ class TestFunctionalAnalysisTest(unittest.TestCase):
         cog_tst = stat.stat(expandfile, sinfo=TTEST_SINFO, paired=False, parametric=True, ontology='cog', mode='f', control_group='s2',
                             outfile=None)
         # make sure false is > 0.05 and trues are less than 0.05
+        cog_tst.set_index('id', inplace=True)
         self.assertTrue(cog_tst['p_s1_over_s2']['C'] > 0.05)
         self.assertTrue(cog_tst['p_s1_over_s2'][['N', 'D']].le(0.05).all())
 
@@ -92,6 +94,7 @@ class TestFunctionalAnalysisTest(unittest.TestCase):
         ec_tst = stat.stat(expandfile, sinfo=TTEST_SINFO, paired=False, parametric=True, ontology='ec', mode='f', control_group='s2',
                            outfile=tested_file)
         # make sure false is > 0.05 and trues are less than 0.05
+        ec_tst.set_index('id', inplace=True)
         self.assertTrue(ec_tst['p_s1_over_s2']['3.4.11.-'] > 0.05)
         self.assertTrue(ec_tst['p_s1_over_s2'][['3.4.21.70', '1.2.-.-']].le(0.05).all())
 
@@ -103,9 +106,10 @@ class TestTaxonomyAnalysisTest(unittest.TestCase):
         expanded = testfile('expand_taxttest.tab')
         tax_df = expand.expand('t', sinfo=TTEST_SINFO, int_file=int, pep_colname_int='peptide', pep_colname_func='peptide',
                                pep_colname_tax='peptide', data_dir=TEST_DIR, outfile=expanded, tax_file=tax, tax_colname='lca')
-        tax_tst = stat.stat(expanded, sinfo=TTEST_SINFO, paired=False, parametric=False, ontology=None, mode=None, control_group='s2'
+        tax_tst = stat.stat(expanded, sinfo=TTEST_SINFO, paired=False, parametric=False, ontology=None, mode=None, control_group='s2',
                             outfile=None)
         # make sure false is > 0.05 and trues are less than 0.05
+        tax_tst.set_index('id', inplace=True)
         self.assertTrue(tax_tst['p_s1_over_s2'][210] > 0.05)
         self.assertTrue(tax_tst['p_s1_over_s2'][[1496,1870884]].le(0.05).all())
         # also, make sure firmicutes phylum is sum of c difficile and clostridiaceae

